src/portable/portable.c

```diff
@@ -1058,6 +1092,12 @@ static int attach_unit_file(
                 if (fd < 0)
                         return log_debug_errno(fd, "Failed to create unit file '%s': %m", path);
 
+                if (!isempty(m->selinux_file_con)) {
```

Comment:
hmm, sounds like something where to use mac_selinux_create_file_prepare(), no? so that the file is created under the right context rather than with a wrong context just to be corrected laterr.

Reply:
I am told that this is already the status quo, as that's what happens when you create a file in a directory by default - but files under a directory may have labels that are valid and allowed but different from the default for that directory. When we do a normal file copy (eg: portablectl attach pointed to a directory instead of an image) the label is transferred correctly, but here it's lost and you can't recover it. Our selinux people told me explicitly to do this to fix the use case. I do not understand selinux, so I just do what the selinux maintainer says verbatim.

Reply:
Ah, sorry I was confused, I actually meant to say `setfscreatecon_raw()`, i.e. it's just a way how you tell the kernel that some label should be used to create the next files *before* creating them. It's usually preferable to create the inode properly labelled, instead of creating it wrongly labelled and then adjusting.

(`mac_selinux_create_file_prepare()` uses that internally too, but consults the selinux db for the right label, and that#s of course not what you want to use here, you already know the label after all)

Reply:
I had looked at that, but given we use O_TMPFILE so the file doesn't actually exist anywhere until it's renamed later on, I thought it wouldn't matter?

Reply:
anyway it's easy enough to add a new helper, fixed